### PR TITLE
Read RI values from comments field into RI field & added unit test

### PR DESCRIPTION
**Pull request by wverastegui**
Classes were adapted to read RI comment field from CSV and MSP input files. Unit test were added accordingly. Also fixed issues related to absence of RT column in CSV file and RT key in the MSP file.

closes https://github.com/RECETOX/RIAssigner/issues/107, closes https://github.com/RECETOX/RIAssigner/issues/110, closes https://github.com/RECETOX/RIAssigner/issues/108

### CODE REVIEW

File: RIAssigner/data/MatchMSData.py [hunk 1]
```diff
@@ -116,6 +116,38 @@ def retention_indices(self, values: Iterable[Data.RetentionIndexType]):
...
+        The key's value converted to string or 'None'.
+    """
+
+    meta_value = spectrum.get(key, default=None)
+    if meta_value is not None:
+        try:
+            meta_value = str(meta_value)
+        except ValueError:
+            meta_value = None
+    return meta_value
```
Comment (hechth) [inline RIAssigner/data/MatchMSData.py:149]: This is not needed as we expect a string to be present here

File: RIAssigner/data/MatchMSData.py [hunk 2]
```diff
@@ -153,4 +184,6 @@ def _assign_ri_value(spectrum: Spectrum, key: str, value: Data.RetentionIndexTyp
     """
     if value is not None:
         retention_index = ('%f' % float(value)).rstrip('0').rstrip('.')
+        if key == 'retentionindex':
+            key = 'retention_index'
         spectrum.set(key=key, value=retention_index)
```
Comment (hechth) [inline RIAssigner/data/MatchMSData.py:189]: Why is this needed?

File: RIAssigner/data/Data.py [hunk 3]
```diff
@@ -115,3 +117,36 @@ def retention_indices(self, value: Iterable[RetentionIndexType]):
...
+    @abstractmethod
+    def comment(self) -> Iterable[CommentFieldType]:
+        """Getter for `comment` property.
+
+        Returns:
+            Iterable[CommentFieldType]: Comment field values stored in data.
+        """
+        ...
+
+    def extract_ri_from_comment(self, specific_string):
```
[the remark this answers is not in the document]
Comment (wverastegui) [reply RIAssigner/data/Data.py]: Done ([7a845bb](https://github.com/RECETOX/RIAssigner/pull/112/commits/7a845bb5fe64abed680620d18cf57f300cadfe6c))

File: RIAssigner/data/MatchMSData.py (hunk 1 above)
Comment (wverastegui) [reply RIAssigner/data/MatchMSData.py]: Addressed [b5244e3](https://github.com/RECETOX/RIAssigner/pull/112/commits/b5244e30132f4c4a30df1efe3b983873e937bdba)

File: RIAssigner/data/MatchMSData.py (hunk 2 above)
Comment (wverastegui) [reply RIAssigner/data/MatchMSData.py]: It is not needed. It was supposed to fix an error that appeared before unit test  implementation. Reverted to original ([f90fd71](https://github.com/RECETOX/RIAssigner/pull/112/commits/f90fd712eae831b7a43c393096a557bc81a35ea4))

**Review by hechth — APPROVED** [review]

File: RIAssigner/data/Data.py [hunk 4]
```diff
@@ -115,3 +117,36 @@ def retention_indices(self, value: Iterable[RetentionIndexType]):
...
+        ----------
+        content_comment:
+            Comment field of the data file. 
+        ri_source:
+            String that is expected to be in the comment field before the RI value.
+
+        Returns
+        -------
+            RI values as a list.
+        """
```
Comment (hechth) [inline RIAssigner/data/Data.py:147]:
```suggestion
    def extract_ri_from_comment(self, ri_source: str):
        """ Extract RI from comment field.
        Extracts the RI from the comment field of the data file. The RI is expected to be
        in the format 'ri_source=RI_value'. The function extracts the RI value and
        sets it on the retention_index property.

        Parameters
        ----------
        content_comment:
            Comment field of the data file. 
        ri_source:
            String that is expected to be in the comment field before the RI value.
        """

```
